k8s/pkg/apis/longhorn/v1beta2/objectstore.go

```diff
@@ -0,0 +1,175 @@
+package v1beta2
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ObjectStoreState string
+
+const (
+	ObjectStoreStateUnknown  = ObjectStoreState("Unknown")
+	ObjectStoreStateStarting = ObjectStoreState("Starting")
+	ObjectStoreStateRunning  = ObjectStoreState("Running")
+	ObjectStoreStateStopping = ObjectStoreState("Stopping")
+	ObjectStoreStateError    = ObjectStoreState("Error")
+)
+
+type ObjectStoreStatus struct {
+	// The state of the object store as observed by the object store controller.
+	//
+	// The object store implemets a state machine with this, beginning at
+	// "unknown". Once the object store controller detects the new object store
+	// and begins resource creation the state is transformed to "starting". The
+	// object store controller observes the resources and once all resources are
+	// up and ready the state transitions to "running".
+	// The state remains in "running" until the object store is deleted, at
+	// which point the controller will clean up the associated resources.
+	//
+	//  │ object store created
+	//  │
+	//  └──►┌───────────┐
+	//      │ unknown   │
+	//  ┌── └───────────┘
+	//  │
+	//  │ controller creates resources
+	//  │
+	//  └──►┌───────────┐
+	//      │ starting  │
+	//  ┌── └───────────┘
+	//  │
+	//  │ controller detects all resources ready
+	//  │
+	//  └──►┌───────────┐
+	//      │ running   │
+	//  ┌── └───────────┘
+	//  │
+	//  │ object endpoint is marked for deletion
+	//  │
+	//  └──►┌───────────┐
+	//      │ stopping  │
+	//  ┌── └───────────┘
+	//  │
+	//  │ controller waits for dependent resources to disappear before removing
+	//  │ the finalizer, thereby letting the object store be deleted
+	//
+	// +optional
+	// +default="Unknown"
+	State ObjectStoreState `json:"state"`
+
+	// A list of addresses where the S3 store is exposed.
+	//
+	// +optional
+	Endpoints []string `json:"endpoints"`
+}
+
+type ObjectStoreCredentials struct {
```

Comment:
Propose to add omitempty instruction. The data over here is being send to backup target. We don't want to store empty field. Ref: https://github.com/longhorn/longhorn-manager/pull/2135/files#diff-2c3ffdbae066b34c3db294fb12d0e69acecd87a77f1100f8789bd885064d46ecR72-R77

Reply:
cc @m-ildefons